src/bitcoin-chainstate.cpp

```diff
@@ -0,0 +1,263 @@
+// The bitcoin-chainstate executable serves to surface the dependencies required
+// by a program wishing to use Bitcoin Core's consensus engine as it is right
+// now.
+//
+// It is part of the libbitcoinkernel project.
+
+#include <iosfwd> // for cout and shit
+#include <functional> // for std::function
+#include <filesystem>
+
+#include <node/chainstate.h>          // for node::LoadChainstate
+#include <init/common.h>              // for SetGlobals, UnsetGlobals
+#include <validation.h>               // for ChainstateManager, InitScriptExecutionCache, StopScriptCheckWorkerThreads, UpdateUncommittedBlockStructures, BlockManager, DEFAULT_CHECKBLOCKS, DEFAULT_CHECKLEVEL
+#include <validationinterface.h>      // for GetMainSignals, cs_main, CMainSignals, RegisterSharedValidationInterface, UnregisterSharedValidationInterface, CValidationInterface
+#include <core_io.h>                  // for DecodeHexBlk
+#include <consensus/validation.h>     // for BlockValidationState
+#include <chainparams.h>              // for Params, SelectParams, CChainParams
+#include <node/blockstorage.h>        // for node::fReindex
+#include <util/system.h>              // for gArgs, ArgsManager
+#include <util/thread.h>              // for TraceThread
+#include <scheduler.h>                // for CScheduler
+#include <script/sigcache.h>          // for InitSignatureCache
+
+    const std::function<std::string(const char*)>
+        G_TRANSLATION_FUN = nullptr;
+
+// Adapted from rpc/mining.cpp
+class submitblock_StateCatcher final : public CValidationInterface
+{
+public:
+    uint256 hash;
+    bool found;
+    BlockValidationState state;
+
+    explicit submitblock_StateCatcher(const uint256 &hashIn) : hash(hashIn), found(false), state() {}
+
+protected:
+    void BlockChecked(const CBlock& block, const BlockValidationState& stateIn) override {
+        if (block.GetHash() != hash)
+            return;
+        found = true;
+        state = stateIn;
+    }
+};
+
+
+int main(int argc, char* argv[]) {
+    if (argc != 2) {
+        std::cerr
+            << "Usage: " << argv[0] << " DATADIR" << std::endl
+            << "Display DATADIR information, and process hex-encoded blocks on standard input." << std::endl;
```

Comment:
Could clarify that this is experimental, for testing only, and expected to break in future versions?

Otherwise someone might accidentally pass the configure option and then open a bug that their mainnet utxo database is corrupted.

Reply:
Fixed as of: f0f78cb20b49f0a022bf3661538db17eb58ad280